### DISCUSSION TITLE
Not getting Code parameter

### DISCUSSION THREAD

**Discussion by MurtazaChohan** [timdorr/tesla-api#382] [Q&A]
I am facing an issue in Integrating tesla unofficial Api
https://tesla-api.timdorr.com/api-basics/authentication
In Step 2
I am getting the response 302 but not getting { code
![Screenshot_20210521_203826](https://user-images.githubusercontent.com/61612693/119163109-88dba200-ba74-11eb-8fa2-edac91a4c184.jpg)
} parameter in Post Api.

**Comment by 2of3Bs** [comment]
I also encountered this just a few days ago after my auth code expired. 
Can anyone confirm if this is the symptom you would see from the Captcha changes?
There were some posts saying the captcha had been removed again, so is this different?

**Reply by mcolyer** [reply]
I am also seeing this - I'm not sure how to fix it.

**Reply by nikemeal** [reply]
I'm getting exactly the same.  Have tried with my own code and also pasting in the other answers in various discussions here and all seem to get back the exact same location URL as the one previously (with client_id/challenge, etc) but have yet to see one with Code=

**Comment by nikemeal** [comment]
I've been having the same issue recently, and a couple of things caused it:
- I'd set my email for `identity` incorrect (I put `,.co.uk`) so the redirect back to the same page was actually it telling me that I'd put incorrect login details. So maybe check the actual HTML of the body coming back to see if it's saying that
- The cookie I got back from the previous step is returned as three separate `set-cookie` headers, but if you just parse headers chances are it will overwrite them so you only ever get the last one.  What I had to do was check if a header with the key already existed and if so append the next one after a space (or implode them, or whatever).  So you have just one string that is made up of the three set-cookie headers.  Then set this as one of the request headers, and also if using curl i set it as the actual cookie (`curl_setopt($ch, CURLOPT_COOKIE, $cookie);`

This now gets me to the multifactor page so I know the login worked so hopefully it might be something similar for others
